### PR TITLE
Add API documentation

### PR DESCRIPTION
**Pull request by sverhoeven**
PR adds docstrings to the public API of the package. 
Also adds Sphinx documentation hosted at readthedocs.org.

API docs are hosted on https://howfairis.readthedocs.io/en/api-docs-51/apidocs/howfairis.html

Refs #51

After PR has been merged the api-docs-51 branch can be removed from  https://howfairis.readthedocs.io

### CODE REVIEW

File: howfairis/Compliance.py [hunk 1]
```diff
@@ -3,6 +3,15 @@
...
 # pylint: disable=too-many-arguments
 class Compliance:
+    """Compliance of a repo against the 5 FAIR software recommendations
+
+    Attributes:
+        repository: Whether repository is publicly accessible with version control
+        license: Whether repository has a license
+        registry: Whether code is in a registry
+        citation: Whether software is citable
+        checklist: Whether a software quality checklist is used
```
Comment (eriktks) [inline howfairis/Compliance.py:13]: `license` is missing a trailing `_` and `compliant_symbol` and `noncompliant_symbol` are missing in the list of attributes

**Review by eriktks — APPROVED** [review]
Checked if code is working and it was fine. Read all new docstrings. Added a comment to the docstring in `howfairis/Compliance.py` about an incorrect name and missing attributes. Rest was fine.

File: howfairis/Compliance.py (hunk 1 above)
Comment (sverhoeven) [reply howfairis/Compliance.py:13]: Incorrect as the argument has `license_`, but attribute of the class `self.license` does not.

Added missing attribs.

**Comment by sverhoeven** [comment]
Thanks for reviewing

Due to merging #134 before this one I needed to do some additional changes so the type hints would render correctly.

**Pull request merged**